test/physics/em/Rayleigh.test.cc

```diff
@@ -33,46 +46,151 @@ class RayleighInteractorTest : public celeritas_test::InteractorHostTestBase
+
+        // Setup MaterialView
+        MaterialParams::Input inp;
+        inp.elements  = {{8, AmuMass{15.999}, "O"},
+                        {74, AmuMass{183.84}, "W"},
+                        {82, AmuMass{207.2}, "Pb"}};
+        inp.materials = {
+            {1.0 * constants::na_avogadro,
+             293.0,
+             celeritas::MatterState::solid,
+             {{celeritas::ElementId{0}, 0.5},
+              {celeritas::ElementId{1}, 0.3},
+              {celeritas::ElementId{2}, 0.2}},
+             "PbWO"},
+        };
+        this->set_material_params(inp);
+        this->set_material("PbWO");
+
+        // Construct RayleighModel and set the host data group
+        model_ = std::make_shared<RayleighModel>(
+            ModelId{0}, particles, *this->material_params());
+        group_ = model_->host_group();
     }
 
     void sanity_check(const Interaction& interaction) const
     {
         ASSERT_TRUE(interaction);
 
-        // Check change to parent track
-        EXPECT_GT(this->particle_track().energy().value(),
+        // Check change to parent track - coherent scattering
+        EXPECT_EQ(this->particle_track().energy().value(),
                   interaction.energy.value());
-        EXPECT_LT(0, interaction.energy.value());
-        EXPECT_SOFT_EQ(1.0, celeritas::norm(interaction.direction));
         EXPECT_EQ(celeritas::Action::scattered, interaction.action);
-
-        // XXX Check secondaries
-
-        // Check conservation between primary and secondaries
-        this->check_conservation(interaction);
     }
 
   protected:
-    celeritas::detail::RayleighInteractorPointers pointers_;
+    std::shared_ptr<RayleighModel>       model_;
+    celeritas::detail::RayleighNativeRef group_;
 };
 
 //---------------------------------------------------------------------------//
 // TESTS
 //---------------------------------------------------------------------------//
 
-TEST_F(RayleighInteractorTest, basic) {}
+TEST_F(RayleighInteractorTest, basic)
+{
+    // Sample an element (TODO: add ElementSelector)
+    ElementId el_id{0};
+
+    std::vector<double>            angle;
+    std::vector<unsigned long int> rng_counts;
+
+    // Sample scattering angle and count rng used for each incident energy
+    for (double inc_e : {0.05, 0.1, 1.0, 10.0, 100.0})
+    {
+        // Set the incident particle energy
+        this->set_inc_particle(pdg::gamma(), MevEnergy{inc_e});
+
+        // Create the interactor
+        RayleighInteractor interact(this->model_->host_group(),
+                                    this->particle_track(),
+                                    this->direction(),
+                                    el_id);
+
+        RandomEngine& rng_engine = this->rng();
+
+        // Produce a sample from the original/incident photon
+        celeritas::Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+        rng_engine.count();
```

Comment:
I think this line is a null-op, and at the beginning/end of the loop you want:
```
        rng_engine.reset_count();
```

Reply:
Okay, I thought that instantiating rng inside loop automatically reset the counter (but actually not).  I move the rng outside the loop and put reset_count at the beginning of the loop - I think that resetting count at the end is redundant for this case.

Reply:
That's a good idea... that would make this much simpler (in a couple of other tests as well). Let me add this to the to-do Q3 review list, and we'll implement after this gets merged.